spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubReactiveAutoConfiguration.java

```diff
@@ -43,17 +47,20 @@
 @ConditionalOnProperty(value = "spring.cloud.gcp.pubsub.reactive.enabled", matchIfMissing = true)
 public class GcpPubSubReactiveAutoConfiguration {
 
-	@Bean
-	@ConditionalOnMissingBean(name = "pubSubReactiveScheduler")
-	Scheduler pubSubReactiveScheduler() {
-		return Schedulers.elastic();
-	}
+	private Scheduler defaultPubSubReactiveScheduler = Schedulers.newElastic("pubSubReactiveScheduler");
 
 	@Bean
 	@ConditionalOnMissingBean
 	public PubSubReactiveFactory pubSubReactiveFactory(
-		PubSubSubscriberTemplate subscriberTemplate, @Qualifier("pubSubReactiveScheduler") Scheduler scheduler) {
-		return new PubSubReactiveFactory(subscriberTemplate, scheduler);
+			PubSubSubscriberTemplate subscriberTemplate,
+			@Qualifier("pubSubReactiveScheduler") Optional<Scheduler> scheduler) {
```

Comment:
Does `Optional<Scheduler>` work for this? I thought you needed to use `ObjectProvider`, but this is cool.

Reply:
Yep, it works. Even declaring something `@Nullable` should work the same way.
(I actually just did it automatically because it would work in Guice, and... it worked.)